configure.ac

```diff
@@ -60,7 +60,6 @@ AC_TYPE_UINT64_T
 #
 # Checks for library functions.
 #
-AC_FUNC_FORK
```

Comment:
I applied your patch but it's not working.
I also tried with
AM_CONDITIONAL(HAVE_FORK, test "$ac_cv_func_fork_works" = "yes")
but it doesn't work in cross-compilation.
Unless I'm missing something, AC_FUNC_FORK automatically creates pre-processor macros that can be evaluated in .c files, not in configure.
https://www.gnu.org/software/autoconf/manual/autoconf-2.60/html_node/Particular-Functions.html

It seems to me that AC_CHECK_FUNC is more appropriate for our purpose.

AC_CHECK_FUNC (function, [action-if-found], [action-if-not-found])

https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Generic-Functions.html

Reply:
Sigh! Cross compiling :/ It tries to guess....here output for a ARM system with MMU:

```
checking for working fork... cross
configure: WARNING: result yes guessed because of cross compilation
checking for working vfork... (cached) yes
```

with the fixed test `AM_CONDITIONAL(HAVE_FORK, test "$ac_cv_func_fork_works" = "yes")` it compile the code that uses `fork()`. What happens with a no-MMU toolchain? Still guessing and deciding for yes?

Reply:
Sorry, but I got confused.
The correction AM_CONDITIONAL(HAVE_FORK, test "$ac_cv_func_fork_works" = "yes") works.
When I was replying to you, I saw that the patch had not been applied.

This is the output with the patch applied:
...
checking for fork... no
checking for vfork... yes
checking for alarm... yes
checking for working vfork... (cached) yes
...

>>> can-utils 2023.03 Building
PATH="/home/dario/projects/passgat/stm32f769i-disco/buildroot/output/host/bin:/home/dario/projects/passgat/stm32f769i-disco/buildroot/output/host/sbin:/home/dario/.local/bin:/home/dario/.local/bin:/home/dario/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin"  /usr/bin/make -j9  -C /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/build/can-utils-2023.03/
  CC       asc2log.o
  CC       lib.lo
  CC       canframelen.lo
  CC       candump.o
  CC       canbusload.o
  CC       canfdtest.o
  CC       cangen.o
  CC       cangw.o
  CC       canplayer.o
  CC       cansend.o
  CC       cansequence.o
  CC       cansniffer.o
  CC       isotpdump.o
  CC       isotpperf.o
  CC       isotprecv.o
  CC       isotpsend.o
  CC       isotpsniffer.o
  CC       isotptun.o
  CC       j1939acd.o
  CC       libj1939.lo
  CC       j1939cat.o
j1939acd.c: In function ‘main’:
j1939acd.c:489:38: warning: passing argument 1 of ‘asprintf’ from incompatible pointer type [-Wincompatible-pointer-types]
  489 |                         if (asprintf(&program_invocation_name, "%s.%s",
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~
      |                                      |
      |                                      const char **
In file included from j1939acd.c:16:
/home/dario/projects/passgat/stm32f769i-disco/buildroot/output/host/arm-buildroot-uclinux-uclibcgnueabi/sysroot/usr/include/stdio.h:375:40: note: expected ‘char ** restrict’ but argument is of type ‘const char **’
  375 | extern int asprintf (char **__restrict __ptr,
      |                      ~~~~~~~~~~~~~~~~~~^~~~~
  CC       j1939spy.o
  CC       j1939sr.o
  CC       log2asc.o
  CC       log2long.o
  CC       slcan_attach.o
  CC       slcand.o
  CC       slcanpty.o
  CC       testj1939.o
  CC       calc-bit-timing/can-calc-bit-timing.o
  CC       mcp251xfd/mcp251xfd-dev-coredump.o
  CC       mcp251xfd/mcp251xfd-dump.o
  CC       mcp251xfd/mcp251xfd-main.o
  CC       mcp251xfd/mcp251xfd-regmap.o
  CCLD     libj1939.la
  CCLD     libcan.la
  CCLD     j1939acd
  CCLD     j1939cat
  CCLD     j1939spy
  CCLD     j1939sr
  CCLD     testj1939
  CCLD     asc2log
  CCLD     can-calc-bit-timing
  CCLD     canbusload
  CCLD     candump
  CCLD     canfdtest
  CCLD     cangen
  CCLD     cangw
  CCLD     canplayer
  CCLD     cansend
  CCLD     cansequence
  CCLD     cansniffer
  CCLD     isotpdump
  CCLD     isotpperf
  CCLD     isotprecv
  CCLD     isotpsend
  CCLD     isotpsniffer
  CCLD     isotptun
  CCLD     log2asc
  CCLD     log2long
  CCLD     mcp251xfd-dump
  CCLD     slcan_attach
  CCLD     slcand
  CCLD     slcanpty
>>> can-utils 2023.03 Installing to target
PATH="/home/dario/projects/passgat/stm32f769i-disco/buildroot/output/host/bin:/home/dario/projects/passgat/stm32f769i-disco/buildroot/output/host/sbin:/home/dario/.local/bin:/home/dario/.local/bin:/home/dario/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin"  /usr/bin/make -j9 DESTDIR=/home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target install -C /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/build/can-utils-2023.03/
make[3]: Nothing to be done for 'install-data-am'.
 /usr/bin/mkdir -p '/home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin'
  /bin/bash ./libtool   --mode=install /usr/bin/install -c asc2log can-calc-bit-timing canbusload candump canfdtest cangen cangw canplayer cansend cansequence cansniffer isotpdump isotpperf isotprecv isotpsend isotpsniffer isotptun j1939acd j1939cat j1939spy j1939sr log2asc log2long mcp251xfd-dump slcan_attach slcand slcanpty testj1939 '/home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin'
libtool: install: /usr/bin/install -c asc2log /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/asc2log
libtool: install: /usr/bin/install -c can-calc-bit-timing /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/can-calc-bit-timing
libtool: install: /usr/bin/install -c canbusload /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/canbusload
libtool: install: /usr/bin/install -c candump /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/candump
libtool: install: /usr/bin/install -c canfdtest /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/canfdtest
libtool: install: /usr/bin/install -c cangen /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/cangen
libtool: install: /usr/bin/install -c cangw /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/cangw
libtool: install: /usr/bin/install -c canplayer /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/canplayer
libtool: install: /usr/bin/install -c cansend /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/cansend
libtool: install: /usr/bin/install -c cansequence /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/cansequence
libtool: install: /usr/bin/install -c cansniffer /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/cansniffer
libtool: install: /usr/bin/install -c isotpdump /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/isotpdump
libtool: install: /usr/bin/install -c isotpperf /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/isotpperf
libtool: install: /usr/bin/install -c isotprecv /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/isotprecv
libtool: install: /usr/bin/install -c isotpsend /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/isotpsend
libtool: install: /usr/bin/install -c isotpsniffer /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/isotpsniffer
libtool: install: /usr/bin/install -c isotptun /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/isotptun
libtool: install: /usr/bin/install -c j1939acd /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/j1939acd
libtool: install: /usr/bin/install -c j1939cat /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/j1939cat
libtool: install: /usr/bin/install -c j1939spy /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/j1939spy
libtool: install: /usr/bin/install -c j1939sr /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/j1939sr
libtool: install: /usr/bin/install -c log2asc /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/log2asc
libtool: install: /usr/bin/install -c log2long /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/log2long
libtool: install: /usr/bin/install -c mcp251xfd-dump /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/mcp251xfd-dump
libtool: install: /usr/bin/install -c slcan_attach /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/slcan_attach
libtool: install: /usr/bin/install -c slcand /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/slcand
libtool: install: /usr/bin/install -c slcanpty /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/slcanpty
libtool: install: /usr/bin/install -c testj1939 /home/dario/projects/passgat/stm32f769i-disco/buildroot/output/target/usr/bin/testj1939

So, I will push with this correction.